src/modules/logicModule/paymentClient/ERC20PaymentClient.sol

```diff
@@ -208,4 +184,53 @@ abstract contract ERC20PaymentClient is
             revert Module__ERC20PaymentClient__InvalidRecipient();
         }
     }
+
+    //--------------------------------------------------------------------------
+    // {ERC20PaymentClient} Function Implementations
+
+    /// @dev Ensures `amount` of payment tokens exist in address(this).
+    function _ensureTokenBalance(uint amount) internal virtual {
```

Comment:
Since this involves a low-level call to an unknown token, we should assume that it could be an ERC-777 token as well which is capable of re-entrancy.

Consider using a ReEntrancy guard on this function?